c/tree.c

```diff
@@ -979,9 +982,9 @@ LOCAL thawed_network_s *thaw_network(MMDBW_tree_s *tree, uint8_t **buffer)
     uint8_t prefix_length = thaw_uint8(buffer);
 
     if (0 == start_ip && 0 == prefix_length) {
-        uint8_t *maybe_separator = thaw_bytes(buffer, strlen(FREEZE_SEPARATOR));
-        if (memcmp(maybe_separator, FREEZE_SEPARATOR,
-                   strlen(FREEZE_SEPARATOR)) == 0) {
+        size_t separator_length = strlen(FREEZE_SEPARATOR);
+        uint8_t *maybe_separator = thaw_bytes(buffer, separator_length + 1);
```

Comment:
Are you sure this is right. When it freezes the separator I don't think it actually freezes the final NUL byte, no there's no NUL byte to thaw.